server/services/store/sqlstore/board.go

```diff
@@ -684,6 +684,52 @@ func (s *SQLStore) searchBoardsForUser(db sq.BaseRunner, term, userID string) ([
 	return s.boardsFromRows(rows)
 }
 
+// searchBoardsForUserInTeam returns all boards that match with the
+// term that are either private and which the user is a member of, or
+// they're open, regardless of the user membership.
+// Search is case-insensitive.
+func (s *SQLStore) searchBoardsForUserInTeam(db sq.BaseRunner, teamID, term, userID string) ([]*model.Board, error) {
```

Comment:
This is almost completely a copy of `SearchBoardsForUser` in MM auth layer. Can one function re-use the other one as the only difference is including team id check?

Reply:
This is for both store and app layers.